### ISSUE TITLE
🐛 invalid char boundary in diagnostics

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#3120]
### Environment information

```block
CLI:
  Version:                      1.8.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "foot"
  JS_RUNTIME_VERSION:           "v22.2.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.8.1"

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

This is an error I extracted from #342
Here a reproduction:

```shell
❯ cat ./1058S11.8.3_A107522.js
if (eval("1\u0009\u000B\u000C\u0020\u00A0\u000A\u000D\u2028\u𪋲029>\u0009\u000B\u000C\u0020\u00A0\u000A\u000D\u2028\u20290") !== true) {

❯ npx biome lint ./1058S11.8.3_A107522.js
./1058S11.8.3_A107522.js:1:61 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ invalid digits after unicode escape sequence
  
  > 1 │ if (eval("1\u0009\u000B\u000C\u0020\u00A0\u000A\u000D\u2028\u𪋲029>\u0009\u000B\u000C\u0020\u00A0\u000A\u000D\u2028\u20290") !== true) {
      │                                                             Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_diagnostics/src/display/frame.rs:242:57
Thread Name: main
Message: byte index 62 is not a char boundary; it is inside '𪋲' (bytes 61..65) of `if (eval("1\u0009\u000B\u000C\u0020\u00A0\u000A\u000D\u2028\u𪋲029>\u0009\u000B\u000C\u0020\u00A0\u000A\u000D\u2028\u20290") !== true) {`
```

### Expected result

Biome should not crash

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
The issue is fixed.
